### DISCUSSION TITLE
EAF toggling command

### DISCUSSION THREAD

**Discussion by c1-g** [emacs-eaf/emacs-application-framework#797] [Show and tell]
So I've been using eaf for its pdf-viewer app since it's faster than pdf-tools but,  I tend to return to pdf-tools for some of its features like incremental searching, occur-mode and its snappy annotation. So for me, juggling between the two can be quite a hassle. This command 
``` emacs-lisp
  (defun eaf-toggle ()
    (interactive)
    (if (advice-member-p 'eaf--find-file-advisor 'find-file)
        (advice-remove #'find-file #'eaf--find-file-advisor)
      (advice-add #'find-file :around #'eaf--find-file-advisor)))
```
will remove eaf advice to `find-file` , disabling eaf as a result until you call it again. I recommend binding it to some convenient key.

**Comment by manateelazycat** [comment]
Why not add pdf to eaf-find-file-ext-blacklist?

**Reply by c1-g** [reply]
Yeah, but then if I wanted to use pdf-viewer again, I would have to remove it from the variable so I recommend doing as you said for more of a permanent kind of thing.

**Reply by manateelazycat** [reply]
You can binding a key to eaf-open comnand, then you don't need switch
